### ISSUE TITLE
图9分割任务的t-SNE图实现

### ISSUE THREAD

**Issue by huangjin520** [VinyehShaw/HFF#4]
作者您好🤩🤩🤩，您这篇工作对问题的分析和实验的构建太精彩啦🤩，特别是后面对网络的视觉可解释性分析，精读完之后深受启发✨。

关于图9 t-SNE我觉得非常有意义，想找您请教一个问题。t-SNE一般是分类任务常用的，虽然分割任务是像素级别的分类，但是会因为像素量大导致无法计算，想知道您制作t-SNE的思路。

期待您的回复，非常感谢，再次感谢您精彩的Paper!😍

**Comment by VinyehShaw** [comment]
Hi there, thank you for your kind words and interest in our work.

Regarding your question about the t-SNE visualization, segmentation is essentially a dense, pixel-level classification task. While the logic is similar to image-level classification, the massive number of pixels in 3D MRI volumes does indeed make direct computation infeasible. For this work, we extract features from the penultimate layer (before the final classifier) rather than the final logits, and apply class-balanced random sampling to make the computation feasible.

This kind of analysis is common and considered standard in many prior segmentation works, (e.g., [CVPR23](https://openaccess.thecvf.com/content/CVPR2023/papers/Yang_Directional_Connectivity-Based_Segmentation_of_Medical_Images_CVPR_2023_paper.pdf#:~:text=This%20paper%20proposed%20a%20novel,shared%20latent%20space%20and%20use), [MICCAI24](https://papers.miccai.org/miccai-2024/paper/1593_paper.pdf), [CVPR24](https://openaccess.thecvf.com/content/CVPR2024/papers/Niu_Unsupervised_Universal_Image_Segmentation_CVPR_2024_paper.pdf), [CVPR24(2)](https://openaccess.thecvf.com/content/CVPR2024/papers/Chen_Mind_Marginal_Non-Crack_Regions_Clustering-Inspired_Representation_Learning_for_Crack_Segmentation_CVPR_2024_paper.pdf)).


Hope this helps clarify your doubts. Good luck with your research. :)

**Comment by huangjin520** [comment]
Hi, **PhD Shao**😊, have a nice day! Thank you very much for your detailed and helpful explanation.

I carefully went through the segmentation papers you mentioned and now understand that workflow for t-SNE visualization in segmentation tasks.

I have one further, more detailed question if you don’t mind.🤩🤩🤩
In practice, is the t-SNE constructed from **a single image, or by the entire dataset**?

Thank you very much again!✨✨✨

**Comment by VinyehShaw** [comment]
Hi, in my opinion, both approaches are acceptable depending on your visualization purpose. Single images highlight local discrimination, while subdataset sampling demonstrates global robustness.
Btw, I am actually still a student, but thanks for the title.

**Comment by huangjin520** [comment]
Thanks again! Have a nice day.
